src/types/interfaces.ts

```diff
@@ -7,3 +7,51 @@ export type RequestApiOptions = {
   authToken?: string;
   logger?: any
 };
+
+export interface ISuperagentNetworkError extends Error {
+  status: number | undefined; // will be `undefined` for network errors & timeouts
+  method: string;
+  url: string;
+  crossDomain: boolean;
+}
+
+export interface ISuperagentResponseError {
+  original?: any;
+  status: number;
+  response: {
+    body: any; // body of the failure response
+    error: ISuperagentNetworkError; // original error thrown by superagent
+    header: { [key: string]: string };
+    headers: { [key: string]: string };
+    status: number;
+    statusCode: number;
+    statusText: string;
+    text: string; // body of response as a string
+    req: { // request
+      method: string;
+      _data?: string; // body of request as a string
+    }
+  };
+}
+
+export interface INetworkError extends IError {
+  status: number;
+  method: string;
+  url: string;
+  crossDomain: boolean;
+}
+
+export interface IResponseError extends IError {
+  status: number;
+  correlationId: string;
+  responseBody: string;
+  method: string;
+  requestBody: string;
+  url: string;
+}
+
+export interface IError {
+  message: string;
+  name: string;
+  stack?: string;
```

Comment:
If the caller sends these errors straight up to sumo, it gets pretty ugly. But I didn't want to just remove the stack altogether from the returned object.